Library/OcCpuLib/OcCpuLib.c

```diff
@@ -107,6 +88,161 @@ ScanThreadCount (
     }
   }
 
+  return Status;
+}
+
+STATIC
+EFI_STATUS
+ScanFrameworkMpServices (
+  IN  FRAMEWORK_EFI_MP_SERVICES_PROTOCOL  *FrameworkMpServices,
+  OUT OC_CPU_INFO                         *Cpu,
+  OUT UINTN                               *NumberOfProcessors,
+  OUT UINTN                               *NumberOfEnabledProcessors
+  )
+{
+  EFI_STATUS           Status;
+  UINTN                Index;
+  EFI_MP_PROC_CONTEXT  Context;
+  UINTN                ContextSize;
+
+  ASSERT (FrameworkMpServices != NULL);
+  ASSERT (Cpu != NULL);
+  ASSERT (NumberOfProcessors != NULL);
+  ASSERT (NumberOfEnabledProcessors != NULL);
+
+  ContextSize = sizeof (Context);
```

Comment:
Please initialize variables as close to the point where they are needed as reasonable, see below. (1)